src/select.rs

```diff
@@ -0,0 +1,160 @@
+                    .short("t")
+                    .value_name("FORMAT")
+                    .takes_value(true)
+                    .possible_values(&["finalfusion", "text", "textdims", "word2vec"])
+                    .default_value("finalfusion")
+                    .help("Output format"),
+            )
+            .arg(
+                Arg::with_name(INPUT_EMBEDDINGS)
+                    .help("Input embeddings")
+                    .index(1)
+                    .required(true),
+            )
+            .arg(
+                Arg::with_name(OUTPUT_EMBEDDINGS)
+                    .help("Output embeddings")
+                    .index(2)
+                    .required(true),
+            )
+            .arg(Arg::with_name(SELECT).help("Words to select").index(3))
+    }
+
+    fn parse(matches: &ArgMatches) -> Result<Self> {
+        let input_filename = matches.value_of(INPUT_EMBEDDINGS).unwrap().to_owned();
+        let output_filename = matches.value_of(OUTPUT_EMBEDDINGS).unwrap().to_owned();
+        let select_input = Input::from(matches.value_of("SELECT"));
+
+        let input_format = matches
+            .value_of(INPUT_FORMAT)
+            .map(|f| {
+                EmbeddingFormat::try_from(f)
+                    .context(format!("Cannot parse embedding format: {}", f))
+            })
+            .transpose()?
+            .unwrap();
+
+        let output_format = matches
+            .value_of(OUTPUT_FORMAT)
+            .map(|f| {
+                EmbeddingFormat::try_from(f)
+                    .context(format!("Cannot parse embedding format: {}", f))
+            })
+            .transpose()?
+            .unwrap();
+
+        Ok(SelectApp {
+            input_filename,
+            input_format,
+            output_filename,
+            output_format,
+            select_input,
+        })
+    }
+
+    fn run(&self) -> Result<()> {
+        let embeddings = read_embeddings(&self.input_filename, self.input_format)
+            .context("Cannot read embeddings")?;
+
+        let select = self
+            .select_input
+            .buf_read()
+            .context("Cannot open selection file")?
+            .lines()
+            .collect::<Result<HashSet<String>, _>>()?;
+
+        let output_embeddings = copy_select_embeddings(&embeddings, select)?;
+
+        write_embeddings(
+            &output_embeddings,
+            &self.output_filename,
+            self.output_format,
+            true,
+        )
+    }
+}
+
+fn copy_select_embeddings(
+    embeddings: &Embeddings<VocabWrap, StorageWrap>,
+    select: HashSet<String>,
+) -> Result<Embeddings<VocabWrap, StorageWrap>> {
+    let mut selected_vocab = Vec::new();
+    let mut selected_storage = Array2::zeros((select.len(), embeddings.dims()));
+    let mut selected_norms = Array1::zeros((select.len(),));
+
+    for (idx, word) in select.into_iter().enumerate() {
+        if !embeddings.embedding_into(&word, selected_storage.row_mut(idx)) {}
```

Comment:
I'm a bit confused by this line, the embedding gets copied into the new storage or it doesn't get copied to it and no matter the result nothing happens?

Reply:
Whoops, that's a leftover from some previous code. Removed it!